generator/dart/generated/google_cloud_functions_v2/lib/cloudfunctions.dart

```diff
@@ -279,7 +279,7 @@ class Function$ extends Message {
       buildConfig: decode(json['buildConfig'], BuildConfig.fromJson),
       serviceConfig: decode(json['serviceConfig'], ServiceConfig.fromJson),
       eventTrigger: decode(json['eventTrigger'], EventTrigger.fromJson),
-      state: decode(json['state'], Function$$State.fromJson),
+      state: decode(json['state'], Function$State.fromJson),
       updateTime: decode(json['updateTime'], Timestamp.fromJson),
       labels: (json['labels'] as Map?)?.cast(),
       stateMessages: decodeList(json['stateMessages'], StateMessage.fromJson),
```

Comment:
(The, fx, ` if (buildConfig != null) 'buildConfig': buildConfig!.toJson(),` could use a pattern to avoid the extra `!`.
`if (buildConfig case final value?) 'buildConfig': value.toJson(),`. Can use longer names than `value`, but it shouldn't matter.)

Reply:
I'm going to mostly re-write the json decoding logic. There are a few corner cases that we need to handle, and I understand it well enough now that I can simplify both the code generation for this as well as the generated code (mostly using specific functions to decode from json and have the code generator just select the right function for the data type).